pkg/agent/openflow/client.go

```diff
@@ -352,6 +394,9 @@ func (c *client) InstallNodeFlows(hostname string,
 			// tunnelPeerIP is the Node Internal Address. In a dual-stack setup, whether this address is an IPv4 address or an
 			// IPv6 one is decided by the address family of Node Internal Address.
 			flows = append(flows, c.l3FwdFlowToRemote(localGatewayMAC, *peerPodCIDR, tunnelPeerIP, cookie.Node))
+		} else if runtime.IsWindowsPlatform() && !c.encapMode.NeedsRoutingToPeer(tunnelPeerIP, c.nodeConfig.NodeIPAddr) && remoteGatewayMAC != nil {
```

Comment:
To divide and conquer, how about just making two cases in this function: encap and noEncap?

- For encap, leave it as it.
- For noEncap, use a platform specific function, for example "l3FwdFlowToRemoteViaRouting".

Then in l3FwdFlowToRemoteViaRouting,

- for Linux version, it's same as before.
- for windows version, if remoteGatewayMAC isn't present, it's same as before; otherwise it uses the new flows.

Reply:
Updated.